src/controller/java/src/matter/controller/Interactions.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ *   Copyright (c) 2023 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+package matter.controller
+
+import kotlinx.coroutines.flow.Flow
+
+interface Interactions {
+  /**
+   * Subscribes to periodic updates of all attributes and events from a device.
+   *
+   * @param request The Subscribe command's path and arguments.
+   * @return A Flow of SubscriptionState representing the subscription's state updates.
+   */
+  fun subscribe(request: SubscribeRequest): Flow<SubscriptionState>
+
+  /**
+   * Issues a read request to a target device for specified attributes and events.
+   *
+   * @param request Read command's path and arguments.
+   * @return A response to the read request.
+   * @throws Generic Exception if an error occurs during the read operation.
+   */
+  suspend fun read(request: ReadRequest): ReadResponse
```

Comment:
Spec don't have anything called readResponse, in this regard, I would prefer ReportResponse or ReportDataResponse instead of ReadResponse.
https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/data_model/Interaction-Model.adoc#report-data-action

Reply:
ReadInteraction does not return the result synchronously; instead, the ReadResponse in this function is returned synchronously which is better reflect its meaning from an API perspective.